delphin/extra/latex.py

```diff
@@ -47,59 +47,77 @@ def dmrs_tikz_dependency(xs):
+\\depstyle{icons}{edge below, dashed}
+\\providecommand{\\spred}{} %% may be defined in mrs.sty
+\\renewcommand{\\spred}[1]{\\mbox{\\textsf{#1}}}
+\\providecommand{\\srl}{} %% may be defined in mrs.sty
+\\renewcommand{\\srl}[1]{\\mbox{\\textsf{\\smaller #1}}}
+%%%
+\\begin{document}""".split("\n")
+    
     for x in xs:
-        lines.append(
-            '\\begin{dependency}['
-            'text only label,'
-            'label style={above,font=\\footnotesize},'
-            'edge unit distance=2ex'
-            ']'
-        )
+        lines.append("\\begin{dependency}[dmrs]")
         ns = nodes(x)
         ls = links(x)
-        predlist = [_latex_escape(n.pred.short_form()) for n in ns]
-        lines.extend([
-            '  \\begin{deptext}[column sep=10pt]',
-            '    {} \\\\'.format(' \\& '.join(predlist)),
-            '  \\end{deptext}'
-        ])
+        ### predicates
+        lines.append("  \\begin{deptext}[column sep=10pt]")
+        for i, n in enumerate(ns):
+            if i < len(ns) -1:
+                sep = "\\&"
+            else:
+                sep = "\\\\"
+            lines.append("    \\spred{{{}}} {}     % node {}".format(_latex_escape(n.pred.short_form()),
+                                                                  sep, i+1))
+        lines.append("  \\end{deptext}")
         nodeidx = {n.nodeid: i+1 for i, n in enumerate(ns)}
+        ### links
         for link in links(x):
             if link.start == 0:
                 lines.append(
-                    '  \\deproot[edge unit distance=2ex]{{{}}}{{{}}}'.format(
+                    '  \\deproot[root]{{{}}}{{{}}}'.format(
                         nodeidx[link.end],
-                        'TOP'  # _latex_escape('/' + link.post)
+                        '\\srl{TOP}'  # _latex_escape('/' + link.post)
                     )
                 )
             else:
-                side = label_side(link)
-                opts = [side]
-                slot = link.end * -1 if side.endswith('below') else link.end
-                if slot in slots[link.start]:
-                    opts.append('edge unit distance=4ex')
-                slots[link.start].add(slot)
-                lines.append('  \\depedge[{}]{{{}}}{{{}}}{{{}}}'.format(
-                    ','.join(opts),
+                # slot = link.end * -1 if side.endswith('below') else link.end
+                # if slot in slots[link.start]:
+                #     opts.append('edge unit distance=4ex')
+                # slots[link.start].add(slot)
+                lines.append('  \\depedge[{}]{{{}}}{{{}}}{{\\srl{{{}}}}}'.format(
+                    label_edge(link),
                     nodeidx[link.start],
                     nodeidx[link.end],
                     _latex_escape(link_label(link))
                 ))
-        lines.append('\\end{dependency}')
+        ### placeholder for icons
+        lines.append('%  \\depedge[icons]{f}{t}{focus}')
```

Comment:
I put it there to make it easy for someone to add icons by hand in the tex if they want to.  
So it should stay in the python, commented out in the tex (where it is easy to uncomment).  
But I will capitalize FOCUS to be more consistent.